### ISSUE TITLE
[Suggestion] Far Future Style

### ISSUE THREAD

**Issue by QuantiX999** [TeamSpen210/BEE2-items#589]
What about the future Aperture Science? Maybe in the parallel universe, where it hasn't collapsed.
Neon blue chamber signs, glowing walls/panels, creating cubes in front of players, with another words:
_**Thinking With Time Machine Style**_

**Comment by TheDarkBomber** [comment]
Technically, Clean style takes place in the far future. And so does Original Clean, which also takes place in parallel universe. But, if GLaDOS didn't fail, and the events of Portal 1 didn't take place, then this would work.

**Comment by Luke18033** [comment]
I just downloaded TWTM, I've found that i't PeTI (no BEE2) only looks like TWTM style in the entrances and exits, the actual chamber just uses Clean. This would be really useful along with TWTM support.

**Comment by ZombiePwner1** [comment]
TWTM Disappointed me in the PeTI when the actual chamber just used regular clean textures if TWTM Style does get added in bee2 make the whole entire map futuristic not just the entrances and exit.

**Comment by TeamSpen210** [comment]
It needs a custom compiler to change wall textures.

**Comment by ZombiePwner1** [comment]
That should be doable shouldn't it?

**Comment by TeamSpen210** [comment]
Of course, that's what we have now for any of the styles. The actual problem is the time-travel/shadow mechanic - implementing it requires modifications to many items, like cubes, buttons, triggers, etc.

**Comment by ZombiePwner1** [comment]
Is it possible to make the time travel shadow mechanic?

**Comment by ellingtonvengaboysmuseum** [comment]
this style could also have a time freeze item unique to only this style implemented...

**Comment by TheDarkBomber** [comment]
@ellingtonvengaboysmuseum No, no items will be unique to a single style. Plus, a time freeze item is impossible to implement in BEE2.4 anyway.

**Comment by Byzarru** [comment]
This style will need a new decorative item as well for the bright parts of the walls.

**Comment by Endermage77** [comment]
I'd imagine that we wouldn't even need to implement the time mechanic into BEE2 itself if it's going to support TWTM, right?

**Comment by TeamSpen210** [comment]
I’d still need to modify about every item to accommodate it.

**Comment by Endermage77** [comment]
Oh.

**Comment by Luke18033** [comment]
Some notes on this:
- TWTM entry/exit doors have larger embed spaces to fit the elevator room. This is a huge style compatibility issue, so we'll want to offer an alternate version which connects the room using worldportals like in other styles, possibly with corridors.
- Cube buttons should be switched to use the normal entity with its model changed, instead of recreating it with triggers (which results in cubes "sticking" to the button).
- On a related note, the "gravity buttons" seen in the campaign would be replaced by #1336.
- For cube droppers, we'd need to add floor/wall variants in other styles to ensure compatibility (#1276).
- Cube builders should be added as a separate item for all styles (#1305), with the original item becoming a normal dropper.

**Comment by Luke18033** [comment]
Also, the style should be usable with any game (though probably still requiring TWTM to be installed, to pack the assets from). This means we would also need Coop entrances and exits for the style. The time machine itself would behave like the paint gun, only being usable in TWTM.

**Comment by Luke18033** [comment]
Additionally there are some bugs we'll want to fix:
- "Magnetic" cube buttons (fixed by using the normal entity)
- Various cube dropper issues
	- Frankenturrets do not spawn, and are replaced by standard cubes when dropperless
	- Companion cubes also don't exist, duplicating the standard cube
	- Spheres and reflection cubes spawn in a different orientation than the assemble animation shows
